website/client/src/components/chat/autoComplete.vue

```diff
@@ -128,38 +130,29 @@ export default {
         backgroundColor: 'white',
       };
     },
-    searchResults () {
-      if (!this.searchActive) return [];
-      if (!this.atRegex.exec(this.text)) return [];
-      this.currentSearch = this.atRegex.exec(this.text)[0]; // eslint-disable-line vue/no-side-effects-in-computed-properties, max-len, prefer-destructuring
-      this.currentSearch = this.currentSearch.substring(1, this.currentSearch.length); // eslint-disable-line vue/no-side-effects-in-computed-properties, max-len
-
-      const lowerCaseSearch = this.currentSearch.toLowerCase();
-      return this.tmpSelections
-        .filter(option => { // eslint-disable-line arrow-body-style
-          return option.displayName.toLowerCase().indexOf(lowerCaseSearch) !== -1
-            || (option.username
-              && option.username.toLowerCase().indexOf(lowerCaseSearch) !== -1);
-        })
-        .slice(0, 4);
-    },
-
   },
   watch: {
-    text (newText) {
-      if (!newText[newText.length - 1] || newText[newText.length - 1] === ' ') {
+    text (newTextParam, prevTextParam) {
+      const delCharFocusBool = prevTextParam.length > newTextParam.length;
+      const caretPosition = this.textbox.selectionEnd;
+      const charFocus = (
+        delCharFocusBool ? prevTextParam : newTextParam
+      )[caretPosition - (delCharFocusBool ? 0 : 1)];
+      if (
+        newTextParam.length === 0 // Delete all
+        || (charFocus === ' ' && !delCharFocusBool) // End Search
```

Comment:
Is it enough to check on space characters for search end?  What about other types of whitespace characters?  Also I don't think the negative deletion check is necessary for spaces, at that point the search should have ended already anyway right?

```suggestion
        || /\s/.test(lastFocussed) // End Search
```

Reply:
Relic code I was copying, will implement this regex.